jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -37,6 +39,36 @@ def put(self, key: str, value: bytes):
     """Adds new cache entry, possibly evicting older entries."""
     if not key:
       raise ValueError("key cannot be empty")
-      #TODO(colemanliyah):implement eviction
-    with open(os.path.join(self._path, key), "wb") as file:
+    if self._evict_entries_if_necessary(key, value):
+      self._create_a_file(self._path, key, value)
+    else:
+      warnings.warn("This compilation file is larger than the size of the cache")
+
+  def _evict_entries_if_necessary(self, key: str, value: bytes) -> bool:
+    new_file_size = len(value)
+
+    if new_file_size >= self._max_cache_size_bytes:
+      return False
+
+    while new_file_size + self._get_cache_directory_size() > self._max_cache_size_bytes:
+      last_time = float('inf')
+      file_to_delete = None
+      for file_name in os.listdir(self._path):
+        file_to_inspect = os.path.join(self._path, file_name)
+        atime = os.stat(file_to_inspect).st_atime
+        if atime < last_time:
+          last_time = atime
+          file_to_delete = file_to_inspect
+      assert file_to_delete
+      os.remove(file_to_delete)
+    return True
+
+  def _create_a_file(self, directory: str, key: str, value: bytes):
```

Comment:
This function is so small and simple now, and only called in one place, that I would copy/paste it directly where you call it and get rid of this function. (Just FYI, I think `_get_cache_directory_size` should still be its own helper function because even though it's only one line, it's pretty fancy logic that merits its own function to describe what its doing.)